CONTRIBUTING.md

```diff
@@ -53,3 +53,92 @@ To build a local image use the `bin/task docker-local` command in the [FerretDB]
 To use a pre-built image you must set the `FERRETDB_IMAGE` environment variable, e.g. `export FERRETDB_IMAGE=ghcr.io/ferretdb/ferretdb-dev:main`.
 
 As mentioned above, this approach is not recommended.
+
+## How to write diff tests
+
+Diff tests demonstrate how FerretDB differs from MongoDB.
+
+We write diff tests when we can't write "regular" integration or compat tests because FerretDB behaves differently
+from MongoDB in some cases.
+If such differences are visible to users, in addition to diff tests, they must be documented
+in the [FerretDB documentation](https://docs.ferretdb.io/diff/),
+please take a look at it for the additional context.
+
+Diff tests are located in the `tests/diff` directory and are regular Go tests.
+
+Currently, every diff test or subtest should have at least two subtests - `FerretDB` and `MongoDB`.
+Those subtests usually only contain asserts based on the expected behavior of each database.
+
+Let's take a look at the following example:
+
+```go
+func TestDollarSign(t *testing.T) {
+    t.Run("Insert", func(t *testing.T) {
+        _, err := db.Collection("collection").InsertOne(ctx,  bson.D{{"foo$", "bar"}})
+
+        t.Run("FerretDB", func(t *testing.T) {
+            expected := mongo.CommandError{
+                Code:    2,
+                Name:    "BadValue",
+                Message: `invalid key: "foo$" (key must not contain '$' sign)`,
+            }
+            AssertEqualError(t, expected, err)
+        })
+
+        t.Run("MongoDB", func(t *testing.T) {
+            require.NoError(t, err)
+        })
+    })
+
+    /* Further subtests */
+}
+```
+
+It works with document validation feature for the `foo$` field key
+and demonstrates how FerretDB validation differs from MongoDB behavior.
+In case of FerretDB a document is considered  invalid if it has a key containing the `$` sign.
+In case of MongoDB such document is considered valid.
+The subtest `Insert` demonstrates this difference: in case of FerretDB we expect to receive a particular error,
+in case of MongoDB we expect to receive no error when we insert such document.
+
+### Configuration
```

Comment:
That section is not specific for diff tests and could be generalized

Reply:
fixed